mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2353,6 +2353,16 @@ LogicalResult ExpandShapeOp::verify() {
            << " dynamic dims while output_shape has " << getOutputShape().size()
            << " values";
 
+  // Verify if provided output shapes are in agreement with output type.
+  DenseI64ArrayAttr staticOutputShapes = getStaticOutputShapeAttr();
+  ArrayRef<int64_t> resShape = getResult().getType().getShape();
```

Comment:
Do we need to check the rank of `resShape` match the rank of   `staticOutputShapes`?

Reply:
That is already checked[ here](https://github.com/llvm/llvm-project/blob/02ce8227ac28e0b83cf780716ae8f912d076eebe/mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp#L2342).